Sources/TSCBasic/Process.swift

```diff
@@ -210,45 +218,47 @@ public final class Process: ObjectIdentifierProtocol {
     public let workingDirectory: AbsolutePath?
 
     /// The process id of the spawned process, available after the process is launched.
-  #if os(Windows)
+    #if os(Windows)
     private var _process: Foundation.Process?
     public var processID: ProcessID {
         return DWORD(_process?.processIdentifier ?? 0)
     }
-  #else
+    #else
     public private(set) var processID = ProcessID()
-  #endif
+    #endif
+
+    // process execution mutable state
+    private var state: State = .idle
 
-    /// If the subprocess has launched.
-    /// Note: This property is not protected by the serial queue because it is only mutated in `launch()`, which will be
-    /// called only once.
-    public private(set) var launched = false
+    /// Lock to protect execution state
+    private let stateLock = Lock()
 
     /// The result of the process execution. Available after process is terminated.
+    /// This will block while the process is running, as such equivalent to `waitUntilExit`
+    @available(*, deprecated, message: "use waitUntilExit instead")
```

Comment:
👀

Reply:
I think being able to get at the execution result can be independently useful.

Reply:
you can get it by using "waitUntilExit" which describes the behavior better. accessing the "result" will block until the result is available, so basically the same as "waitUntilExit"

Reply:
Using `waitUntilExit` makes sense although the name is a little awkward if you're in a context where you know that the process has already completed.  Although in that case presumably the code that originally did the `waitUntilExit` should just pass along the result anyway.

So I don't have a strong opinion here, and can certainly see the argument for making `waitUnilExit` the way you get the result.

Reply:
Sure, my point was if someone is holding onto a process instance, they might want to ask for the result again, independently of `waitUntilExit`.

Reply:
gotcha, let see if we get push back on this deprecation warning. we can always remove it. I do think the semantics here is a bit odd to.